gyp_uv.py

```diff
@@ -34,37 +44,47 @@ def host_arch():
   return machine  # Return as-is and hope for the best.
 
 
-def run_gyp(args):
-  rc = gyp.main(args)
-  if rc != 0:
-    print('Error running GYP')
-    sys.exit(rc)
+def config_vs2017(config_fn):
+  if os.environ.get('VisualStudioVersion') != '15.0': return
+  o = dict()
+  td = o['target_defaults'] = dict()
+  try:
+    td['msbuild_toolset'] = 'v141'
+    sdk_ver = os.environ['WindowsSDKVersion'].rstrip("\\")
+    td['msvs_windows_target_platform_version'] = sdk_ver
+  except Exception as e:
+    print(' Error: could not find a compatible SDK, ' +
+          'make sure to run from a "VS developer command prompt"')
+    sys.exit(1)
+  write(config_fn, o)
 
 
-if __name__ == '__main__':
-  args = sys.argv[1:]
+def run_gyp(args):
 
   # GYP bug.
   # On msvs it will crash if it gets an absolute path.
   # On Mac/make it will crash if it doesn't get an absolute path.
   if sys.platform == 'win32':
-    args.append(os.path.join(uv_root, 'uv.gyp'))
-    common_fn  = os.path.join(uv_root, 'common.gypi')
-    options_fn = os.path.join(uv_root, 'options.gypi')
-    # we force vs 2010 over 2008 which would otherwise be the default for gyp
-    if not os.environ.get('GYP_MSVS_VERSION'):
```

Comment:
Bottom line `GYP_MSVS_VERSION` comes from detection code in `vcbuild.bat`
IMHO here it's dropped out of the blue.